services/src/main/java/org/keycloak/theme/DefaultThemeManager.java

```diff
@@ -276,6 +277,32 @@ private Map<Locale, Properties> getMessagesByLocale(String baseBundlename, Local
             }
         }
 
+        protected void addlocaleTranslations(Locale locale, Properties m) throws IOException {
+            for (String l : getProperties().getProperty("locales", "").split(",")) {
+                l = l.trim();
+                String key = "locale_" + l;
+                String label = m.getProperty(key);
+                if (label == null) {
+                    String rl = l;
+                    if (l.equals("zh-CN")) {
+                        rl = "zh-HANS";
+                    } else if (l.equals("zh-TW")) {
+                        rl = "zh-HANT";
+                    }
```

Comment:
Let's also add a comment here explaining why this is needed, as well as perhaps logging an issue to remove this in the future.

Reply:
will be part of next commit